src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -182,6 +182,113 @@ public virtual void NextBytes(byte[] buffer)
         /// <param name="buffer">The array to be filled with random numbers.</param>
         public virtual void NextBytes(Span<byte> buffer) => _impl.NextBytes(buffer);
 
+        /// <summary>
+        ///   Fills the elements of a specified span with items chosen at random from choices.
+        /// </summary>
+        /// <param name="choices">The items to use to populate the span.</param>
+        /// <param name="destination">The span to be filled with items.</param>
+        /// <typeparam name="T">The type of span.</typeparam>
+        /// <exception cref="ArgumentException"><paramref name="choices" /> is empty.</exception>
+        /// <remarks>
+        ///   The method uses <see cref="Next(int)" /> to select items randomly from <paramref name="choices" />
+        ///   by index and populate <paramref name="destination" />.
+        /// </remarks>
+        public void GetItems<T>(ReadOnlySpan<T> choices, Span<T> destination)
+        {
+            if (choices.IsEmpty)
+                throw new ArgumentException(SR.Arg_EmptySpan, nameof(choices));
+
+            for (int i = 0; i < destination.Length; i++)
+            {
+                destination[i] = choices[Next(choices.Length)];
+            }
```

Comment:
This is going to end up making destination.Length virtual calls to Next.  Whether as part of this PR or subsequently, we might want to look at overriding this (or since it's not virtual some private protected method it could delegate to) on each of our internal concrete derived types.

Reply:
There was some discussion about this in the API review and (my take away) was that it was a non-issue. Something about "partial guarded de-virtualization" and, since it's in a loop, the JIT should be smart enough to do the virtual table lookup once.

https://youtu.be/73BAn9B3ni4?t=9737

Reply:
> partial guarded de-virtualization

If this is referring to PGO, I'm not convinced that's going to be super meaningful here. For that to help, the majority use case would need to be always using `new Random()`, `new Random(int)`, or `Random.Shared`, and not using a combination of those in the app, since this method is unlikely to be inlined and so there's only going to be a single copy of the method's code used for all uses, regardless of how the instance was constructed.

It's possible everything will get hoisted, but I'd be a little surprised by that as well, especially in the Random.Shared case, e.g. ideally for Random.Shared the implementation would fetch the [ThreadStatic] XoshiroImpl instance once and use it for all of the calls to that instance.

It'd be worth measuring.  If we want to do it subsequent to this PR, fine.

Reply:
@tannergooding since it was you that made the comment, do you have any additional thought here?

Reply:
I believe Tanner is offline for a while. You can go ahead with what you have and we can revisit it subsequently.